pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -329,18 +329,51 @@ function parseSaveError(err: any) {
 }
 
 /**
- * Provide a mapping from settings version to a function used to update the
- * settings object to the next version.
+ * ReplacementDirective describes how a setting can be migrated.
+ */
+interface ReplacementDirective {
+  /**
+   * The path to the old value
+   */
+  oldPath: string;
+  /**
+   * The path to the new value
+   */
+  newPath: string;
+}
+
+/**
+ * This function takes an array of `ReplacementDirectives`, and carries out each one which essentially
+ * moves a value at an old location to a new one, and then deletes the old location.
+ * @param settings - the settings object
+ * @param replacements - a table used to update the settings object based on existing obsolete fields that need to be moved.
+ */
+function processReplacements(settings: any, replacements: ReplacementDirective[]) {
+  for (const replacement of replacements) {
+    const { oldPath, newPath } = replacement;
+
+    if (_.hasIn(settings, oldPath)) {
+      // Transfer the current value for the old field to the new field
+      _.set(settings, newPath, _.get(settings, oldPath));
+      // Delete the old field
+      _.unset(settings, oldPath);
+    }
+  }
+}
+
+/**
+ * Provide a mapping from settings version X to version X + 1
  *
- * The main use-cases are for renaming property names, correct values that are
- * no longer valid, and removing obsolete entries. The final step merges in
- * current defaults, so we won't need an entry for every version change, as
- * most changes will get picked up from the defaults.
+ * Some migrations need to be done with bespoke code, but most of them
+ * can be expressed in a descriptive table, and the operations are done
+ * by `processReplacements`, which just moves values to new locations,
+ * and deletes the old location.
  */
 export const updateTable: Record<number, (settings: any) => void> = {
   1: (settings) => {
-    // Implement setting change from version 3 to 4
-    if ('rancherMode' in settings.kubernetes) {
+    // EXPLANATION: We don't call `processReplacements` in this step
+    // because this migration only deletes an existing field.
+    if (_.hasIn(settings, 'kubernetes.rancherMode')) {
       delete settings.kubernetes.rancherMode;
     }
```

Comment:
Since we're already using lodash, this is equivalent to:
```suggestion
    _.unset(settings, 'kubernetes.rancherMode');
```

Reply:
My comment didn't get pushed:

I only use `_.unset` either for a dynamic string, or one where we're trying to delete `settings.A.B.…Y.Z` and there's no guarantee that `k...Y` exists for some `k` >= `A`